CIが通ってないことに気づいた。以下のコンパイルエラー修正がtodo.
```
 g++ -g -MMD -MP -O3 -std=c++14 -Wall -Wextra -Werror -Wfatal-errors -Wno-unknown-pragmas -march=native -mtune=native -fopenmp -isystem ../../viennacl -isystem ../../googletest/googletest/include -o testobj/test_ComputerImplicitForces.o -c test_ComputerImplicitForces.cpp
test_ComputerImplicitForces.cpp: In member function ‘bool {anonymous}::OpenMps::ImplicitForcesTest::IsAlive(const {anonymous}::OpenMps::Particle&)’:
test_ComputerImplicitForces.cpp:156:162: error: ‘IsSurface’ is not a member of ‘{anonymous}::OpenMps::Computer<boost::numeric::ublas::c_vector<double, 2> (&)(long unsigned int, double, double), boost::numeric::ublas::c_vector<double, 2> (&)(double, double)>’
     return p.TYPE() != Particle::Type::Dummy && p.TYPE() != Particle::Type::Disabled && !OpenMps::Computer<decltype(positionWall)&, decltype(positionWallPre)&>::IsSurface(p.N(), env.N0(), env.SurfaceRatio);
                                                                                                                                                                  ^~~~~~~~~
compilation terminated due to -Wfatal-errors.
make: *** [testobj/test_ComputerImplicitForces.o] Error 1
Makefile:20: recipe for target 'testobj/test_ComputerImplicitForces.o' failed
make: *** Waiting for unfinished jobs....
##[error]Process completed with exit code 2.
```

--- a/src/OpenMps/test/test_ComputerImplicitForces.cpp
+++ b/src/OpenMps/test/test_ComputerImplicitForces.cpp
@@ -24,8 +24,8 @@ namespace {
 	static constexpr double surfaceRatio = 0.95;
 #endif
 	// 格子状に配置する際の1辺あたりの粒子数
-	static constexpr std::size_t num_x = 10;
-	static constexpr std::size_t num_z = 10;
+	static constexpr std::size_t num_x = 7;
+	static constexpr std::size_t num_z = 7;
 
 	// サイズ上限を 横 2*l0*num_x, 縦 2*l0*num_z とする
 	static constexpr double minX = -l0 * num_x;
@@ -156,6 +156,12 @@ namespace {
 				return p.TYPE() != Particle::Type::Dummy && p.TYPE() != Particle::Type::Disabled && !OpenMps::Computer<decltype(positionWall)&, decltype(positionWallPre)&>::IsSurface(p.N(), env.N0(), env.SurfaceRatio);
 			}
 
+			bool IsSurface(const Particle& p)
+			{
+				const auto& env = computer->GetEnvironment();
+				return OpenMps::Computer<decltype(positionWall)&, decltype(positionWallPre)&>::IsSurface(p.N(), env.N0(), env.SurfaceRatio);
+			}
+
 			auto& getPpe()
 			{
 				return computer->ppe;
@@ -191,6 +197,7 @@ namespace {
 		}
 
 		// 係数行列 a_ii = -Σa_ij (i!=j) という恒等式は成立するか？
+		// 境界から離れた中央粒子においてテスト
 		// (dummy, disable, free surface粒子は除外)
 		TEST_F(ImplicitForcesTest, MatrixDiagIdentity)
 		{
@@ -202,23 +209,19 @@ namespace {
 			const auto Ndim = num_x * num_z;
 
 			const auto& particles = GetParticles();
-			for (auto i = decltype(Ndim){0}; i < Ndim; i++)
-			{
-				if (!IsAlive(particles[i]) || ppe.A(i, i) == 1.0)
-				{
-					continue;
-				}
+			const auto ic = (num_x - 1) / 2 * (num_z + 1); // 中央粒子のindex
 
-				double sum_nondiag = 0.0;
-				for (auto j = decltype(Ndim){0}; j < Ndim; j++)
-				{
-					if (j != i && IsAlive(particles[j]))
+			double sum_nondiag = 0.0;
+			for (auto j = decltype(Ndim){0}; j < Ndim; j++)
+			{
+				if (j != ic) {
+					if (IsAlive(particles[j]))
 					{
-						sum_nondiag += ppe.A(i, j);
+						sum_nondiag += ppe.A(ic, j);
 					}
 				}
-				ASSERT_NEAR(ppe.A(i, i), -sum_nondiag, testAccuracy);
 			}
+			ASSERT_NEAR(ppe.A(ic, ic), -sum_nondiag, testAccuracy);
 		}
 
 		// 粒子i の 近傍粒子j に対応する成分が a_ij != 0 であること
@@ -235,7 +238,7 @@ namespace {
 
 			for (auto i = decltype(Ndim){0}; i < Ndim; i++)
 			{
-				if (!IsAlive(particles[i]))
+				if (!IsAlive(particles[i]) || IsSurface(particles[i]))
 				{
 					continue;
 				}

--- a/src/OpenMps/test/test_ComputerImplicitForces.cpp
+++ b/src/OpenMps/test/test_ComputerImplicitForces.cpp
@@ -192,7 +192,6 @@ namespace {
 
 		// 係数行列 a_ii = -Σa_ij (i!=j) という恒等式は成立するか？
 		// 境界から離れた中央粒子においてテスト
-		// (dummy, disable, free surface粒子は除外)
 		TEST_F(ImplicitForcesTest, MatrixDiagIdentity)
 		{
 			SearchNeighbor();
@@ -208,11 +207,9 @@ namespace {
 			double sum_nondiag = 0.0;
 			for (auto j = decltype(Ndim){0}; j < Ndim; j++)
 			{
-				if (j != ic) {
-					if (IsAlive(particles[j]))
-					{
-						sum_nondiag += ppe.A(ic, j);
-					}
+				if (j != ic)
+				{
+					sum_nondiag += ppe.A(ic, j); // disable,dummy,free surface particleは寄与しない
 				}
 			}
 			ASSERT_NEAR(ppe.A(ic, ic), -sum_nondiag, testAccuracy);

--- a/src/OpenMps/test/test_ComputerImplicitForces.cpp
+++ b/src/OpenMps/test/test_ComputerImplicitForces.cpp
@@ -153,13 +153,7 @@ namespace {
 			bool IsAlive(const Particle& p)
 			{
 				const auto& env = computer->GetEnvironment();
-				return p.TYPE() != Particle::Type::Dummy && p.TYPE() != Particle::Type::Disabled && !OpenMps::Computer<decltype(positionWall)&, decltype(positionWallPre)&>::IsSurface(p.N(), env.N0(), env.SurfaceRatio);
-			}
-
-			bool IsSurface(const Particle& p)
-			{
-				const auto& env = computer->GetEnvironment();
-				return OpenMps::Computer<decltype(positionWall)&, decltype(positionWallPre)&>::IsSurface(p.N(), env.N0(), env.SurfaceRatio);
+				return p.TYPE() != Particle::Type::Dummy && p.TYPE() != Particle::Type::Disabled;
 			}
 
 			auto& getPpe()
@@ -238,7 +232,7 @@ namespace {
 
 			for (auto i = decltype(Ndim){0}; i < Ndim; i++)
 			{
-				if (!IsAlive(particles[i]) || IsSurface(particles[i]))
+				if (!IsAlive(particles[i]))
 				{
 					continue;
 				}

